rust/experimental/query_engine/expressions/src/static_scalar_expressions.rs

```diff
@@ -0,0 +1,149 @@
+use chrono::{DateTime, FixedOffset};
+
+use crate::{Expression, QueryLocation};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum StaticScalarExpression {
```

Comment:
The list of statics is going to grow. Array, Map, Regex. Null? Perhaps things like Json & Xml. I felt it better to have all the statics defined in their own enum.